### PR TITLE
Fixed PropTypes deprecation of React 15.5

### PR DESCRIPTION
**Pull request by oriSomething**
- Moreover, fixed small linting issue of `defaultMatches` not having propType

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,8 @@
     "test": "karma start --single-run"
   },
   "peerDependencies": {
-    "react": ">=15 || ^0.14.7"
+    "prop-types": "^15.5.8",
+    "react": ">=15"
```
Comment (mjackson) [inline package.json]: Why did you remove `|| ^0.14.7`?

File: package.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
     "test": "karma start --single-run"
   },
   "peerDependencies": {
-    "react": ">=15 || ^0.14.7"
+    "prop-types": "^15.5.8",
```
Comment (mjackson) [inline package.json]: Peer dependencies should be kept as loose as possible to make it easy for people using different versions. Let's use `>=15` here, same as `react`.

File: package.json (hunk 1 above)
Comment (oriSomething) [inline package.json]: Since if someone uses `react 14.x` he probably don't install the new `prop-types` package which is needed with the new React.
I preferred not to do something like:

```js
let PropTypes;
try {
  PropTypes = require("prop-types");
} catch(e) {
  PropTypes = require("react").PropTypes;
}

```

**Comment by oriSomething** [comment]
I've copy pasted since Github put it on "outdated":

Since if someone uses `react 14.x` he probably don't install the new `prop-types` package which is needed with the new React.
I preferred not to do something like:

```js
let PropTypes;
try {
  PropTypes = require("prop-types");
} catch(e) {
  PropTypes = require("react").PropTypes;
}

```

File: package.json (hunk 1 above)
Comment (danhayden) [inline package.json]: Based on the `prop-type` [readme](https://www.npmjs.com/package/prop-types#how-to-depend-on-this-package) it looks like `prop-types` should be a dependency and not a peer dependency and it is [compatible](https://www.npmjs.com/package/prop-types#compatibility) with React 14 back to `react@0.14.9` and React 15 back to `react@15.3.0` therefore the react peer dependency could be `"react": ">=15.3.0 || ^0.14.9"`

**Comment by oriSomething** [comment]
@mjackson Is the changes are okay? Is anymore thing to be done?

**Comment by deiucanta** [comment]
@mjackson can you please merge this? it's really annoying to see the warning error from react :)

**Comment by mjackson** [comment]
This was fixed in #66
